multiqc.wdl

```diff
@@ -0,0 +1,85 @@
+version 1.0
+
+task MultiQC {
+    input {
+        String? preCommand
+        File analysisDirectory
+        Array[File] dependencies   # This must be used in order to run multiqc after these tasks.
+        Boolean force = false
+        Boolean dirs = false
+        Int? dirsDepth
+        Boolean fullNames = false
+        String? title
+        String? comment
+        String? fileName
+        String? outDir
+        String? template
+        String? tag
+        String? ignore
+        String? ignoreSamples
+        Boolean ignoreSymlinks = false
+        File? sampleNames
+        File? fileList
+        Array[String]+? exclude
+        Array[String]+? module
+        Boolean dataDir = false
+        Boolean noDataDir = false
+        String? dataFormat
+        Boolean zipDataDir = false
+        Boolean export = false
+        Boolean flat = false
+        Boolean interactive = false
+        Boolean lint = false
+        Boolean pdf = false
+        Boolean megaQCUpload = false # This must be actively enabled in my opinion. The tools default is to upload.
+        File? config  # A directory
+        String? clConfig
+        Boolean verbose  = false
+        Boolean quiet = false
+    }
+
+    String outputDir = if defined(outDir) then select_first([outDir]) else "."
+
+    command {
+        set -e -o pipefail
+        ~{preCommand}
+        ~{if defined(outDir) then "mkdir -p " + outputDir else ""}
+        multiqc \
+        ~{true="--force" false="" force} \
+        ~{true="--dirs" false="" dirs} \
+        ~{"--dirs-depth " + dirsDepth} \
+        ~{true="--fullnames" false="" fullNames} \
+        ~{"--title " + title} \
+        ~{"--comment " + comment} \
+        ~{"--filename " + fileName} \
```

Comment:
Is this file created? It isn't listed in the outputs.

Reply:
Based on the line just above the output section, you expect this to end in `.html`, though, so it isn't really a prefix?

Reply:
Multiqc parses this in a complex way. If it ends in html, html is cut off. If it doesn't nothing is cut off. This makes output file name unstable and compensation for that is needed.

Reply:
Fair enough